app/uk/gov/hmrc/vatsubscription/models/ReturnPeriod.scala

```diff
@@ -16,28 +16,26 @@
 
 package uk.gov.hmrc.vatsubscription.models
 
-import play.api.Logger
 import play.api.libs.json._
 
-sealed trait ReturnPeriod {
+abstract class ReturnPeriod(transOrCapEmail: Option[String]) {
   def stdReturnPeriod: String
+  val transactorOrCapacitorEmail = transOrCapEmail
 }
 
-case object MAReturnPeriod extends ReturnPeriod {
+case class MAReturnPeriod(transOrCapEmail: Option[String]) extends ReturnPeriod(transOrCapEmail: Option[String]) {
```

Comment:
In contrast to the above comment, the `transOrCapEmail` vals in the `ReturnPeriod` constructors here do not need a type definition because they already have one specified in the case class constructor.